status-im/status-react
# Pairing & multi device & group-chats

### Pairing

Add a new section (under advanced) where you can pair new devices.
I have included a commit from a different PR as both rely on a status-go update and probably best to be tested together.

The code is behind a feature flag but enabled in nighties if development mode is on.
The UI is not polished and this is only the first step, but I decided to enable it in nightlies in dev-mode as it can already provide value to whoever wants to use it.

It is only used to sync up your devices contacts. Group chats will also receive your own messages once devices are paired correctly.

### What's left

1) Polish UI
2) Let the user manually confirm the devices (currently they are enabled without user confirmation)
3) Add information about each device (is this desktop/mobile/last-seen)
4) Sync other kind of info (your name etc)
5) Keep data in sync after initial pairing (contacts, 1-to-1 messages)


### Testing

Both group chats & pairing should only be enabled in nightlies/dev if dev-mode is on.

To pair your devices:

1) Install this build on both desktop & mobile
2) On android/ios go to settings->advanced->devices, on desktop go to Profile->Devices->Pair
3) Pressing `+` will advertise your device. You should now see a new ID in all the `devices` sections.
4) Do the same on the other devices until all your devices shows up in the list.
5) Press on any device ID to share your contacts with all the rest of the devices.
6) Group chats should receive your own messages now, if you start a new group chat with someone, their message should be received by all the paired devices.







status: ready
